core/git_mixins/active_branch.py

```diff
@@ -34,7 +35,8 @@ def get_branch_status(self):
         if first_line.startswith("## Initial commit on "):
             return "Initial commit on `{}`.".format(first_line[21:])
 
-        short_status_pattern = r"## ([A-Za-z0-9\-_\/]+)(\.\.\.([A-Za-z0-9\-_\/]+)( \[((ahead (\d+))(, )?)?(behind (\d+))?\])?)?"
+        branch_pattern = "[A-Za-z0-9" + re.escape(re.sub("[~\\^\\:\\?\\*\\[\"]", "", string.punctuation)) + "]+?"
+        short_status_pattern = r"## (" + branch_pattern + ")(\.\.\.(" + branch_pattern + ")( \[((ahead (\d+))(, )?)?(behind (\d+))?\])?)?$"
```

Comment:
Because of the reluctant quantifier in the branch pattern, we need to add the `$` at the end to make sure the whole line is matched. In the case of

    ## branch_name

With the `$` it yields `branch_name`.
Without the `$` it yields `b`. Very reluctant.